src/FormattedInput.js

```diff
@@ -15,132 +15,133 @@ export const createFormat = (formats, formatChar) => ({
 });
 
 const FormattedInput = ({ value, formatter, onChange, ...props }) => {
+  const [formattedValue, setFormattedValue] = useState(format(formatter)(value));
   const inputEl = useRef(null);
-  const [state, setState] = useState({
+  const stateRefs = useRef({
     selectionStart: 0,
     selectionEnd: 0,
-    rawValue: value,
-    delete: false,
-    formattedValue: format(formatter)(value)
-  });
+    isDelete: false,
+    rawValue: '',
+  })
   useLayoutEffect(() => {
     // A lot of the work here is cursor manipulation
     if (inputEl.current && inputEl.current === document.activeElement) {
       inputEl.current.setSelectionRange(
-        state.selectionStart,
-        state.selectionEnd
+        stateRefs.current.selectionStart,
+        stateRefs.current.selectionEnd
       );
     }
-  });
+  }, [stateRefs]);
+  const handleChange = (event) => {
+    const deleteKeyPressed = stateRefs.current.isDelete;
+    const maxFormatExceeded = stateRefs.current.rawValue.length >= formatter.formats.length - 1;
+    if (maxFormatExceeded && !deleteKeyPressed) {
+      return;
+    }
```

Comment:
If the format length is exceeded we can do an early return.

This inadvertently seems to have fixed this defect where the formatting disappears if you type a large number in this field, and NaN is rendered (discovered on master):

Master:
![Payment Methods](https://github.com/user-attachments/assets/ae22d7e9-3f33-46ee-b5ed-276c2bd90fcf)
New code: 
![Pasted Graphic 13](https://github.com/user-attachments/assets/89010d62-e943-4ac9-bd83-f9a8dbb25c41)